scaaml/capture/input_generators/attack_point_iterator_models.py

```diff
@@ -114,3 +115,55 @@ def get_generated_keys(self) -> List[str]:
 
 BasicIteratorModels: TypeAlias = Union[ConstantIteratorModel,
                                        GeneratedIteratorModel]
+
+
+class RepeatIteratorModel(BaseModel):
+    """
+    Initialize the repeated iterate. If repetitions is not present
+    or set to a negative number it will do an infinite loop and
+    if it is 0 it will not repeat at all.
+          
+        Args:
+            operation (Literal['repeat']): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file.
+
+            configuration (BasicIteratorModels): The config for the iterated
+                object that will get repeated.
+                
+            repetitions (int): This parameter decides how many times the
+                iterator gets repeated. If it is a negative number it
+                will repeat infinitely and if you call __len__ it will
+                raise an LengthIsInfiniteException. If it is 0 then it will not
+                repeat at all. If it is a positive number it will
+                repeat that many times.
+    """
+    operation: Literal["repeat"]
+    repetitions: int = Field(default=-1)
+    configuration: BasicIteratorModels
+
+    @model_validator(mode="after")
+    def check_model(self) -> "RepeatIteratorModel":
+        if len(self.configuration) == 0:
+            self.repetitions = 0
+        return self
+
+    def __len__(self) -> int:
+        if self.repetitions >= 0:
+            return self.repetitions * len(self.configuration)
+        elif len(self.configuration) == 0:
```

Comment:
This branch cannot happen? Because the `check_model` is getting called right after the object is created (i.e., before `__len__` could be called. Also anything times zero is zero (`int` does not have a representation of infinity).